app/javascript/controllers/tooltip_controller.js

```diff
@@ -0,0 +1,82 @@
+import { Controller } from '@hotwired/stimulus'
+import {
+  computePosition,
+  flip,
+  shift,
+  offset,
+  arrow
+} from '@floating-ui/dom';
+
+export default class extends Controller {
+  static targets = ["element", "tooltip"];
```

Comment:
We can remove `element` from this targets array.  By default, Stimulus controllers [set the `this.element` property](https://stimulus.hotwired.dev/reference/controllers#properties) which we can use.

Furthermore, I think we should introduce `arrow` as a target here.  We're heavily depending on the arrow to have an `id` of `arrow` which will have to be extracted when we start introducing multiple tooltips per page.

So the effect of these changes would be:

```js
// Controller

static targets = ["arrow", "tooltip"];

connect() {
  this.element.addEventListener("mouseenter", this.showTooltip)
  this.element.addEventListener("mouseleave", this.hideTooltip)
  this.element.addEventListener("focus", this.showTooltip)
  this.element.addEventListener("blur", this.hideTooltip)
}

disconnect() {
  this.element.removeEventListener("mouseenter", this.showTooltip)
  this.element.removeEventListener("mouseleave", this.hideTooltip)
  this.element.removeEventListener("focus", this.showTooltip)
  this.element.removeEventListener("blur", this.hideTooltip)
}
```

```erb
# View

<div data-tooltip-target="arrow"></div>
```

Reply:
[79f6ab0](https://github.com/maybe-finance/maybe/pull/1065/commits/79f6ab073ce4fc71d24803896450bc98a4f6d4c9)